ui/src/app/utils/user-apps-utils.tsx

```diff
@@ -19,33 +19,46 @@ import { appsApi } from 'app/services/swagger-fetch-clients';
 import { setSidebarActiveIconStore } from 'app/utils/navigation';
 import { userAppsStore } from 'app/utils/stores';
 
-import { GKE_APP_PROXY_PATH_SUFFIX } from './constants';
 import { fetchWithErrorModal } from './errors';
+import { getLastActiveEpochMillis, setLastActive } from './inactivity';
+
+// the polling timeout to use when waiting for a transition (e.g. from Running to Paused)
+const transitionPollingTimeoutMs = 10e3; // 10 sec
+// when we are not waiting for a transition, we still need to poll for activity
+const activityPollingTimeoutMs = 5 * 60e3; // 5 min
 
 export const appTypeToString: Record<AppType, string> = {
   [AppType.CROMWELL]: 'Cromwell',
   [AppType.RSTUDIO]: 'RStudio',
   [AppType.SAS]: 'SAS',
 };
 
-const appStatusesRequiringUpdates: Array<AppStatus> = [
+const transitionalAppStatuses: Array<AppStatus> = [
   AppStatus.DELETING,
   AppStatus.PROVISIONING,
   AppStatus.STARTING,
   AppStatus.STOPPING,
 ];
 
-const doUserAppsRequireUpdates = () => {
-  const { userApps } = userAppsStore.get();
-  return (
-    !userApps ||
-    userApps
-      .map((userApp) => userApp.status)
-      .some((appStatus) => appStatusesRequiringUpdates.includes(appStatus))
+const doUserAppsRequireUpdates = (userApps: ListAppsResponse) =>
```

Comment:
this parameter is available when we call it, so no need to go back to the appsStore